src/main/groovy/JavadocGroupBuilder.groovy

```diff
@@ -49,7 +52,7 @@ public class JavadocGroupBuilder {
 
         def repoUrl = pluginLocation + gid + "/" + id + "/"
         if (version == null) {
-            def metadataURL = repoUrl + "maven-metadata.xml"
+            def metadataURL = "https://repo.jenkins-ci.org/releases/" + gid + "/" + id + "/maven-metadata.xml"
```

Comment:
Compromise here with hardcoded `repo.jenkins-ci.org` to avoid adding a complete download with auth of the metadata to replace `new URL (metadataURL).text` two lines below.

Reply:
Uncompromised.